module/chunks/chunkVerifier.go

```diff
@@ -204,15 +204,42 @@ func (fcv *ChunkVerifier) verifyTransactionsInContext(
 	chunkState := fvmState.NewExecutionState(nil, fvmState.DefaultParameters())
 
 	var problematicTx flow.Identifier
-
 	// collect execution data formatted transaction results
+	var txStartIndex int
+	var systemResult *flow.LightTransactionResult
+
+	if systemChunk {
+		transactions, systemResult, err = fcv.createSystemChunk(
+			callbackCtx,
+			&snapshotTree,
+			chunkState,
+			transactionOffset,
+			&events,
+			&serviceEvents,
+			unknownRegTouch,
+			execResID,
+			chIndex,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create system chunk transactions: %w", err)
+		}
+	}
+
 	var txResults []flow.LightTransactionResult
 	if len(transactions) > 0 {
 		txResults = make([]flow.LightTransactionResult, len(transactions))
 	}
 
-	// executes all transactions in this chunk
-	for i, tx := range transactions {
+	// If system chunk, we already executed the process callback transaction so skip it
+	// by setting the start index to 1 and assigning existing process result to tx results
+	if systemResult != nil {
+		txResults[0] = *systemResult
+		txStartIndex = 1
+	}
+
+	// Executes all transactions in this chunk (or remaining transactions for callbacks)
+	for i := txStartIndex; i < len(transactions); i++ {
+		tx := transactions[i]
```

Comment:
need to fix switching context in next line with callback context when needed - will do on Monday